src/main/java/net/pms/util/FileUtil.java

```diff
@@ -378,14 +415,51 @@ public static String getExtension(String fileName, LetterCase convertTo, Locale
 		return extension;
 	}
 
-	public static String getFileNameWithoutExtension(String f) {
-		int point = f.lastIndexOf('.');
+	/**
+	 * Returns a filename with the extension (if any) stripped off.
+	 *
+	 * @param file the {@link File}.
+	 * @return The extensionless filename.
+	 */
+	public static String getFileNameWithoutExtension(File file) {
+		if (file == null) {
+			return null;
+		}
+		return getFileNameWithoutExtension(file.getName());
+	}
 
-		if (point == -1) {
-			point = f.length();
+	/**
+	 * Returns a filename with the extension (if any) stripped off.
+	 *
+	 * @param file the {@link Path}.
+	 * @return The extensionless filename.
+	 */
+	public static String getFileNameWithoutExtension(Path file) {
+		if (file == null) {
+			return null;
+		}
+		Path fileName = file.getFileName();
+		if (fileName == null) {
+			return null;
 		}
+		return getFileNameWithoutExtension(fileName.toString());
+	}
 
-		return f.substring(0, point);
+	/**
+	 * Returns a filename with the extension (if any) stripped off.
+	 *
+	 * @param fileName the filename.
+	 * @return The extensionless filename.
```

Comment:
Again one new english word learned today ;)

Reply:
@Sami32 Which word? 😕

Reply:
"extensionless"
I didn't knew that it was possible to write it in one word only ;)

Reply:
@Sami32 To be honest, nether did I. In Norwegian you can always combine words in that way, and we do it to a much greater extent than English, so I sometimes "invent" English words that way. I guess I was lucky this time, but I think the meaning should be pretty clear regardless.